zerver/views/realm.py

```diff
@@ -197,47 +197,32 @@ def update_realm(
 
     message_content_delete_limit_seconds: Optional[int] = None
     if message_content_delete_limit_seconds_raw is not None:
-        message_content_delete_limit_seconds = parse_message_content_edit_or_delete_limit(
+        (
+            message_content_delete_limit_seconds,
+            setting_value_changed,
+        ) = parse_and_set_setting_value_if_required(
+            realm,
+            "message_content_delete_limit_seconds",
             message_content_delete_limit_seconds_raw,
-            Realm.MESSAGE_CONTENT_EDIT_OR_DELETE_LIMIT_SPECIAL_VALUES_MAP,
-            setting_name="message_content_delete_limit_seconds",
+            acting_user=user_profile,
         )
-        if (
-            message_content_delete_limit_seconds is None
-            and realm.message_content_delete_limit_seconds is not None
-        ):
-            # We handle 'None' here separately, since in the loop below,
-            # do_set_realm_property is called only if setting value is
-            # not None.
-            do_set_realm_property(
-                realm,
-                "message_content_delete_limit_seconds",
-                message_content_delete_limit_seconds,
-                acting_user=user_profile,
-            )
+
+        if setting_value_changed:
             data["message_content_delete_limit_seconds"] = message_content_delete_limit_seconds
```

Comment:
Just a note that I think this return value code is planned to be deleted in favor of the `ignored_parameters_unsupported` system that @laurynmm was working on as a better way to communicate over the API which of the parts of a request we in fact processed -- but we can do that refactor when it's appropriate.